rust/experimental/query_engine/kql-parser/src/aggregate_expressions.rs

```diff
@@ -0,0 +1,158 @@
+use data_engine_expressions::*;
+use data_engine_parser_abstractions::*;
+use pest::iterators::Pair;
+
+use crate::{Rule, scalar_expression::parse_scalar_expression};
+
+pub(crate) fn parse_aggregate_assignment_expression(
+    aggregate_assignment_expression_rule: Pair<Rule>,
+    state: &ParserState,
+) -> Result<(Box<str>, AggregationExpression), ParserError> {
+    let mut aggregate_assignment_rules = aggregate_assignment_expression_rule.into_inner();
+
+    let destination_rule = aggregate_assignment_rules.next().unwrap();
+    let destination_rule_str = destination_rule.as_str();
+
+    let aggregation_expression =
+        parse_aggregate_expression(aggregate_assignment_rules.next().unwrap(), state)?;
+
+    Ok((destination_rule_str.into(), aggregation_expression))
+}
+
+fn parse_aggregate_expression(
+    aggregate_expression_rule: Pair<Rule>,
+    state: &ParserState,
+) -> Result<AggregationExpression, ParserError> {
+    let query_location = to_query_location(&aggregate_expression_rule);
+
+    let aggregate = match aggregate_expression_rule.as_rule() {
+        Rule::average_aggregate_expression => AggregationExpression::new(
+            query_location,
+            AggregationFunction::Average,
+            Some(parse_scalar_expression(aggregate_expression_rule, state)?),
+        ),
+        Rule::count_aggregate_expression => {
+            AggregationExpression::new(query_location, AggregationFunction::Count, None)
+        }
+        Rule::maximum_aggregate_expression => AggregationExpression::new(
+            query_location,
+            AggregationFunction::Maximum,
+            Some(parse_scalar_expression(aggregate_expression_rule, state)?),
+        ),
+        Rule::minimum_aggregate_expression => AggregationExpression::new(
+            query_location,
+            AggregationFunction::Minimum,
+            Some(parse_scalar_expression(aggregate_expression_rule, state)?),
+        ),
+        Rule::sum_aggregate_expression => AggregationExpression::new(
+            query_location,
+            AggregationFunction::Sum,
+            Some(parse_scalar_expression(aggregate_expression_rule, state)?),
+        ),
+        _ => panic!("Unexpected rule in aggregate_expression: {aggregate_expression_rule}"),
```

Comment:
Not a big fan of `panic!` statements in parsing stuff. We should return an error otherwise killing the system will be fairly easy.